summingbird-online-perf/src/main/scala/com/twitter/summingbird/online/executor/TestUtils.scala

```diff
@@ -0,0 +1,43 @@
+package com.twitter.summingbird.online.executor
+
+import org.openjdk.jmh.infra.Blackhole
+import scala.concurrent.duration.DurationInt
+import scala.util.{Failure, Success}
+
+object TestUtils {
+  val TickDelay = 5.millis
+
+  def testOperationContainer[I, O](
+    container: OperationContainer[I, O, InputState[Int]],
+    inputs: Array[I],
+    bh: Blackhole
+  ): Unit = {
+    var left = inputs.length
+    for (
+      input <- inputs;
+      emitted <- container.execute(InputState[Int](1), input)
+    ) {
+      emitted match {
+        case (states, output) =>
+          states.foreach { _ => left -= 1 }
+          output match {
+            case Success(value) => value.foreach(bh.consume)
+            case Failure(exception) => bh.consume(exception)
+          }
+      }
+    }
+    while (left != 0) {
+      Thread.sleep(TickDelay.toMillis)
```

Comment:
Sorry, I've just noticed that this is actually to check when it's completed. Maybe you could use a `CountDownLatch` for it?